### ISSUE TITLE
Exception: "Invalid attempt to destructure non-iterable instance" when column sorting is enabled

### ISSUE THREAD

**Issue by electrikaLaszlo** [handsontable/handsontable#4563]
### Description
When the columnSorting: true, sortIndicator: true and you want to change the element of the source an exception is thrown: "Invalid attempt to destructure non-iterable instance".  You have to open the console (F12) and you will see the error after 5 second in the demo. UPDATE: The filter option also affect to this issue.
                

### Steps to reproduce

1. Enable the columnSorting: true, sortIndicator: true options
2. Create handsontable with object array source
3. Change an element property in the data source 

### Demo
http://jsfiddle.net/sqcqahxd/

### Your environment
* Handsontable version: 1.14.2
* Browser Name and version: Chrome: Version 61.0.3163.100, Internet Explorer: 11
* Operating System: Windows 10

**Comment by AMBudnik** [comment]
Thank you for sharing @electrikaLaszlo 

I have simplified the code and find out that when we comment `filters` we do not get the console log http://jsfiddle.net/u7ef8u0w/

**Comment by AWulkan** [comment]
Hey, just checking in to see if there's any news related to this bug.
I'm using the latest Handsontable Pro version with the latest Vue wrapper and I still experience this.

Thanks.

**Comment by AMBudnik** [comment]
Sorry @AWulkan. We do not have any updates on this issue yet

**Comment by AWulkan** [comment]
Okey, no problem.
It seems like the bug goes away if I set observeChanges to false in my settings.
I don't know which side effects this might have, but nothing appears to be malfunctioning.

**Comment by AMBudnik** [comment]
Here is one related issue https://github.com/handsontable/handsontable/issues/2685
Also, there was another one but I am not able to find it.

**Comment by AMBudnik** [comment]
Hi @electrikaLaszlo @AWulkan I'm not able to replicate the issue in the latest version. Can you confirm it?

**Comment by AMBudnik** [comment]
I am closing this issue as there is no reply for more than a week but feel free to leave a comment if there's anything wrong with this part of code.

**Comment by AWulkan** [comment]
Sorry for not responding. I haven't had time to try it out and I don't know when I'll be able to. If I notive any problems when I update I'll come back and leave a comment.

**Comment by AMBudnik** [comment]
Thank you @AWulkan

**Comment by AMBudnik** [comment]
Hi @electrikaLaszlo @awulkan 

We just fixed an issue related to yours. Please update to 8.0.0-beta1 and check if it solved for you as well.

Below are the resources

```html
</style><script src="https://cdn.jsdelivr.net/gh/handsontable/handsontable@release/8.0.0-beta.1/dist/handsontable.full.js"></script><link type="text/css" rel="stylesheet" href="https://cdn.jsdelivr.net/gh/handsontable/handsontable@release/8.0.0-beta.1/dist/handsontable.full.css">
```

And this is how you can download the beta from NPM

`npm i handsontable@beta`

**Comment by AMBudnik** [comment]
Referenced issue https://github.com/handsontable/handsontable/issues/2685 closed as fixed for v 8.0.0
